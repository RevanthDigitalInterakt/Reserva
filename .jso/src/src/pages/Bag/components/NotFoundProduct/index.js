  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = NotFoundProduct;
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[1]);
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function NotFoundProduct() {
    var _useBagStore = (0, _$$_REQUIRE(_dependencyMap[2]).useBagStore)(['productNotFound', 'actions']),
      productNotFound = _useBagStore.productNotFound,
      actions = _useBagStore.actions;
    var refs = {
      animatedViewRef: (0, _react.useRef)(null)
    };
    var handleSetNotFoundProduct = (0, _react.useCallback)(function () {
      actions.CLEAR_PRODUCT_NOT_FOUND();
    }, [actions]);
    (0, _react.useEffect)(function () {
      if (refs.animatedViewRef.current) {
        if (refs.animatedViewRef.current.slideInDown) {
          refs.animatedViewRef.current.slideInDown();
        }
      }
    }, [refs.animatedViewRef]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_reactNative.View, {
      ref: refs.animatedViewRef,
      testID: "com.usereserva:id/NotFoundProduct_container",
      animation: "slideInDown",
      useNativeDriver: true,
      style: {
        elevation: 10,
        position: 'absolute',
        right: 0,
        left: 0,
        zIndex: 2
      },
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsxs)(_$$_REQUIRE(_dependencyMap[4]).Box, {
        minHeight: 60,
        bg: "white",
        paddingLeft: "xxxs",
        py: "micro",
        flexDirection: "row",
        alignItems: "center",
        paddingRight: "xxxs",
        boxShadow: _reactNative.Platform.OS === _$$_REQUIRE(_dependencyMap[5]).platformType.IOS ? 'topBarShadow' : null,
        style: {
          elevation: 10
        },
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[4]).Box, {
          flex: 1,
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Typography, {
            fontFamily: "nunitoRegular",
            fontSize: 13,
            color: "preto",
            children: `${productNotFound}`
          })
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[7]).Button, {
          testID: "com.usereserva:id/NotFoundProduct_setProduct",
          flex: 1,
          onPress: handleSetNotFoundProduct,
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[8]).IconLegacy, {
            name: "Close",
            size: 15,
            color: "preto",
            ml: "xxxs"
          })
        })]
      })
    });
  }
