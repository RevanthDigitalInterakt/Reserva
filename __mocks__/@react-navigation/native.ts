const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
