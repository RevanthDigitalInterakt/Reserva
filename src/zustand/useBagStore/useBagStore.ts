import { create } from 'zustand';
import type { IBagStore } from './types/bagStore';
import { createZustandStoreWithSelectors } from '../../utils/createZustandStoreWithSelectors';
import { getApolloClient } from '../../utils/getApolloClient';
import {
  OrderFormAddDiscountCouponDocument,
  OrderFormAddDiscountCouponMutation,
  OrderFormAddDiscountCouponMutationVariables,
  OrderFormAddGiftDocument,
  OrderFormAddGiftMutation,
  OrderFormAddGiftMutationVariables,
  OrderFormAddItemDocument,
  OrderFormAddItemMutation,
  OrderFormAddItemMutationVariables,
  OrderFormAddSellerCouponDocument,
  OrderFormAddSellerCouponMutation,
  OrderFormAddSellerCouponMutationVariables,
  OrderFormDocument,
  OrderFormQuery,
  OrderFormQueryVariables,
  OrderFormRefreshDataDocument,
  OrderFormRefreshDataMutation,
  OrderFormRefreshDataMutationVariables,
  OrderFormRemoveDiscountCouponDocument,
  OrderFormRemoveDiscountCouponMutation,
  OrderFormRemoveDiscountCouponMutationVariables,
  OrderFormRemoveGiftDocument,
  OrderFormRemoveGiftMutation,
  OrderFormRemoveGiftMutationVariables,
  OrderFormRemoveSellerCouponDocument,
  OrderFormRemoveSellerCouponMutation,
  OrderFormRemoveSellerCouponMutationVariables,
  OrderFormRemoveUnavailableItemsDocument,
  OrderFormRemoveUnavailableItemsMutation,
  OrderFormRemoveUnavailableItemsMutationVariables,
  OrderFormSetGiftSizeDocument,
  OrderFormSetGiftSizeMutation,
  OrderFormSetGiftSizeMutationVariables,
  OrderFormUpdateItemDocument,
  OrderFormUpdateItemMutation,
  OrderFormUpdateItemMutationVariables,
} from '../../base/graphql/generated';
import { getAsyncStorageItem } from '../../hooks/useAsyncStorageProvider';
import { getMessageErrorWhenUpdateItem } from './helpers/getMessageErrorWhenUpdateItem';

const bagStore = create<IBagStore>((set, getState): IBagStore => ({
  initialized: false,
  topBarLoading: false,
  loadingModal: false,
  initialLoad: false,
  productNotFound: '',
  error: '',
  deleteProductModal: {
    show: false,
    deleteInfo: undefined,
  },
  //
  orderFormId: '',
  messages: [],
  clientProfileData: undefined,
  items: [],
  selectableGift: undefined,
  marketingData: undefined,
  shippingData: undefined,
  appTotalizers: {
    delivery: 0,
    discount: 0,
    items: 0,
    total: 0,
  },
  installmentInfo: {
    installmentPrice: 0,
    installmentsNumber: 0,
    totalPrice: 0,
  },
  allItemsQuantity: 0,
  hasPrimeSubscriptionInCart: false,
  //
  currentSelectedGiftSize: '',
  currentSelectedColorGift: '',
  actions: {
    INITIAL_LOAD: async () => {
      try {
        if (getState().initialized) return;

        const orderFormId = await getAsyncStorageItem('orderFormId') || '';

        if (!orderFormId) {
          throw new Error('OrderForm inválido.');
        }

        set(() => ({ initialLoad: true }));

        const { data } = await getApolloClient().query<OrderFormQuery, OrderFormQueryVariables>({
          query: OrderFormDocument,
          fetchPolicy: 'no-cache',
          variables: { orderFormId },
          context: { clientName: 'gateway' },
        });

        const { orderForm } = data;

        if (!orderForm) {
          throw new Error('OrderForm inválido.');
        }

        set(() => ({
          orderFormId,
          messages: orderForm.messages,
          clientProfileData: orderForm.clientProfileData,
          items: orderForm.items,
          selectableGift: orderForm.selectableGift,
          marketingData: orderForm.marketingData,
          shippingData: orderForm.shippingData,
          appTotalizers: orderForm.appTotalizers,
          installmentInfo: orderForm.installmentInfo,
          allItemsQuantity: orderForm.allItemsQuantity,
          hasPrimeSubscriptionInCart: orderForm.hasPrimeSubscriptionInCart,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ initialLoad: false, initialized: true }));
      }
    },
    REFETCH_ORDER_FORM: async () => {
      try {
        const orderFormId = await getAsyncStorageItem('orderFormId') || '';

        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().query<OrderFormQuery, OrderFormQueryVariables>({
          query: OrderFormDocument,
          fetchPolicy: 'no-cache',
          variables: { orderFormId },
          context: { clientName: 'gateway' },
        });

        const { orderForm } = data;

        if (!orderForm) {
          throw new Error('OrderForm inválido.');
        }

        set(() => ({
          orderFormId,
          messages: orderForm.messages,
          clientProfileData: orderForm.clientProfileData,
          items: orderForm.items,
          selectableGift: orderForm.selectableGift,
          marketingData: orderForm.marketingData,
          shippingData: orderForm.shippingData,
          appTotalizers: orderForm.appTotalizers,
          installmentInfo: orderForm.installmentInfo,
          allItemsQuantity: orderForm.allItemsQuantity,
          hasPrimeSubscriptionInCart: orderForm.hasPrimeSubscriptionInCart,
          initialized: true,
          topBarLoading: false,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false, initialized: true }));
      }
    },
    REFRESH_ORDER_FORM: async () => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().query<
        OrderFormRefreshDataMutation,
        OrderFormRefreshDataMutationVariables
        >({
          query: OrderFormRefreshDataDocument,
          fetchPolicy: 'no-cache',
          variables: {
            input: { orderFormId: getState().orderFormId },
          },
          context: { clientName: 'gateway' },
        });

        const { orderFormRefreshData: orderForm } = data;

        set(() => ({
          clientProfileData: orderForm.clientProfileData,
          items: orderForm.items,
          marketingData: orderForm.marketingData,
          shippingData: orderForm.shippingData,
          installmentInfo: orderForm.installmentInfo,
          appTotalizers: orderForm.appTotalizers,
          allItemsQuantity: orderForm.allItemsQuantity,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    ADD_SELLER_COUPON: async (sellerCoupon: string) => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().mutate<
        OrderFormAddSellerCouponMutation,
        OrderFormAddSellerCouponMutationVariables
        >({
          mutation: OrderFormAddSellerCouponDocument,
          context: { clientName: 'gateway' },
          variables: {
            coupon: sellerCoupon,
            orderFormId: getState().orderFormId,
          },
        });

        const { orderFormAddSellerCoupon: orderForm } = data || {};

        if (!orderForm) {
          throw new Error('Cupom inválido.');
        }

        set(() => ({
          marketingData: orderForm.marketingData,
          appTotalizers: orderForm.appTotalizers,
          installmentInfo: orderForm.installmentInfo,
          allItemsQuantity: orderForm.allItemsQuantity,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));

        throw new Error(error.message);
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    ADD_DISCOUNT_COUPON: async (coupon: string) => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().mutate<
        OrderFormAddDiscountCouponMutation,
        OrderFormAddDiscountCouponMutationVariables
        >({
          mutation: OrderFormAddDiscountCouponDocument,
          variables: {
            orderFormId: getState().orderFormId,
            coupon,
          },
          context: { clientName: 'gateway' },
        });

        const { orderFormAddDiscountCoupon: orderForm } = data || {};

        const isInvalidCoupon = orderForm?.messages.find((item) => (
          item.includes(coupon) && item.includes('inválido')
        ));

        if (isInvalidCoupon) {
          throw new Error('Cupom inválido.');
        }

        set(() => ({
          marketingData: orderForm?.marketingData,
          appTotalizers: orderForm?.appTotalizers,
          installmentInfo: orderForm?.installmentInfo,
          allItemsQuantity: orderForm?.allItemsQuantity,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));

        throw new Error(error.message);
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    REMOVE_SELLER_COUPON: async () => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().mutate<
        OrderFormRemoveSellerCouponMutation,
        OrderFormRemoveSellerCouponMutationVariables
        >({
          mutation: OrderFormRemoveSellerCouponDocument,
          variables: { orderFormId: getState().orderFormId },
          context: { clientName: 'gateway' },
        });

        const { orderFormRemoveSellerCoupon: orderForm } = data || {};

        set(() => ({
          marketingData: {
            ...getState().marketingData,
            sellerCoupon: '',
            sellerCouponName: '',
          },
          appTotalizers: orderForm?.appTotalizers,
          installmentInfo: orderForm?.installmentInfo,
          allItemsQuantity: orderForm?.allItemsQuantity,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    REMOVE_DISCOUNT_COUPON: async () => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().mutate<
        OrderFormRemoveDiscountCouponMutation,
        OrderFormRemoveDiscountCouponMutationVariables
        >({
          mutation: OrderFormRemoveDiscountCouponDocument,
          variables: { orderFormId: getState().orderFormId },
          context: { clientName: 'gateway' },
        });

        const { orderFormRemoveDiscountCoupon: orderForm } = data || {};

        set(() => ({
          marketingData: {
            ...getState().marketingData,
            coupon: '',
          },
          appTotalizers: orderForm?.appTotalizers,
          installmentInfo: orderForm?.installmentInfo,
          allItemsQuantity: orderForm?.allItemsQuantity,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    ADD_AVAILABLE_GIFT: async (gift, giftId) => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().query<
        OrderFormSetGiftSizeMutation,
        OrderFormSetGiftSizeMutationVariables
        >({
          query: OrderFormSetGiftSizeDocument,
          fetchPolicy: 'no-cache',
          variables: {
            giftId,
            id: gift.id,
            seller: gift.seller,
            orderFormId: getState().orderFormId,
          },
          context: { clientName: 'gateway' },
        });

        const { orderFormSetGiftSize: orderForm } = data || {};

        set({
          marketingData: orderForm.marketingData,
          appTotalizers: orderForm.appTotalizers,
          installmentInfo: orderForm.installmentInfo,
          allItemsQuantity: orderForm.allItemsQuantity,
          items: orderForm.items,
          selectableGift: orderForm.selectableGift,
        });
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    REMOVE_UNAVAILABLE_ITEMS: async () => {
      try {
        set(() => ({ topBarLoading: true }));

        await getApolloClient().mutate<
        OrderFormRemoveUnavailableItemsMutation,
        OrderFormRemoveUnavailableItemsMutationVariables
        >({
          mutation: OrderFormRemoveUnavailableItemsDocument,
          context: { clientName: 'gateway' },
          variables: { orderFormId: getState().orderFormId },
        });
      } catch (error) {
        //
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    UPDATE_PRODUCT_COUNT: async (index, item, countUpdated) => {
      try {
        set(() => ({ topBarLoading: true }));

        let errorsMessages = '';

        const { data } = await getApolloClient().mutate<
        OrderFormUpdateItemMutation,
        OrderFormUpdateItemMutationVariables
        >({
          mutation: OrderFormUpdateItemDocument,
          variables: {
            orderFormId: getState().orderFormId,
            index,
            id: item.id,
            seller: item.seller,
            quantity: item.isAssinaturaSimples ? 1 : countUpdated,
          },
          context: { clientName: 'gateway' },
        });

        const { orderFormUpdateItem: orderForm } = data || {};

        if (orderForm?.messages?.length) {
          errorsMessages = getMessageErrorWhenUpdateItem({
            currentItem: item,
            updateItemResponse: data?.orderFormUpdateItem,
            currentUpdateValueItem: countUpdated,
            appTotalizers: orderForm.appTotalizers,
          });
        }

        set(() => ({
          items: orderForm?.items,
          selectableGift: orderForm?.selectableGift,
          marketingData: orderForm?.marketingData,
          appTotalizers: orderForm?.appTotalizers,
          installmentInfo: orderForm?.installmentInfo,
          allItemsQuantity: orderForm?.allItemsQuantity,
          deleteProductModal: {
            show: false,
            deleteInfo: undefined,
          },
          productNotFound: errorsMessages,
          hasPrimeSubscriptionInCart: orderForm?.hasPrimeSubscriptionInCart,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    ADD_GIFT: async (index, id) => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().mutate<
        OrderFormAddGiftMutation,
        OrderFormAddGiftMutationVariables
        >({
          mutation: OrderFormAddGiftDocument,
          context: { clientName: 'gateway' },
          variables: {
            orderFormId: getState().orderFormId,
            index,
            id,
          },
        });

        set(() => ({ items: data?.orderFormAddGift.items }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    REMOVE_GIFT: async (index, id) => {
      try {
        const { data } = await getApolloClient().mutate<
        OrderFormRemoveGiftMutation,
        OrderFormRemoveGiftMutationVariables
        >({
          mutation: OrderFormRemoveGiftDocument,
          context: { clientName: 'gateway' },
          variables: {
            orderFormId: getState().orderFormId,
            index,
            id,
          },
        });

        set(() => ({ items: data?.orderFormRemoveGift.items }));
      } catch (error) {
        set(() => ({ error: error.message }));
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
    ACTIVE_MODAL_DELETE_PRODUCT: async (product, index) => {
      set(() => ({
        deleteProductModal: {
          show: true,
          deleteInfo: { product, index },
        },
      }));
    },
    CLOSE_MODAL_DELETE_PRODUCT: async () => {
      set(() => ({ deleteProductModal: { show: false, deleteInfo: undefined } }));
    },
    SELECT_GIFT_COLOR: (giftColor) => {
      set(() => ({ currentSelectedColorGift: giftColor }));
    },
    SELECT_GIFT_SIZE: (giftSize) => {
      set(() => ({ currentSelectedGiftSize: giftSize }));
    },
    SELECT_GIFT: (color, size) => {
      set(() => ({
        currentSelectedColorGift: color,
        currentSelectedGiftSize: size,
      }));
    },
    CLEAR_PRODUCT_NOT_FOUND: () => {
      set(() => ({
        productNotFound: '',
      }));
    },
    ADD_ITEM: async (seller, id, quantity) => {
      try {
        set(() => ({ topBarLoading: true }));

        const { data } = await getApolloClient().mutate<
        OrderFormAddItemMutation,
        OrderFormAddItemMutationVariables
        >({
          mutation: OrderFormAddItemDocument,
          context: { clientName: 'gateway' },
          variables: {
            input: {
              orderFormId: getState().orderFormId,
              seller,
              id,
              quantity,
            },
          },
        });

        const { orderFormAddItem: orderForm } = data || {};

        set(() => ({
          items: orderForm?.items,
          selectableGift: orderForm?.selectableGift,
          marketingData: orderForm?.marketingData,
          appTotalizers: orderForm?.appTotalizers,
          installmentInfo: orderForm?.installmentInfo,
          allItemsQuantity: orderForm?.allItemsQuantity,
          deleteProductModal: {
            show: false,
            deleteInfo: undefined,
          },
          hasPrimeSubscriptionInCart: orderForm?.hasPrimeSubscriptionInCart,
        }));
      } catch (error) {
        set(() => ({ error: error.message }));

        throw new Error(error.message);
      } finally {
        set(() => ({ topBarLoading: false }));
      }
    },
  },
}));

export const useBagStore = createZustandStoreWithSelectors(bagStore);
