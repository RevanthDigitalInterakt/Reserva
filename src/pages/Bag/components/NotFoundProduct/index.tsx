import React, { useCallback, useEffect, useRef } from 'react';
import { Platform } from 'react-native';
import { View } from 'react-native-animatable';

import { Box } from '../../../../components/Box/Box';
import { Button } from '../../../../components/Button';
import { IconLegacy } from '../../../../components/IconLegacy/IconLegacy';
import { Typography } from '../../../../components/Typography/Typography';
import { platformType } from '../../../../utils/platformType';
import { useBagStore } from '../../../../zustand/useBagStore/useBagStore';

export default function NotFoundProduct() {
  const { productNotFound, actions } = useBagStore(['productNotFound', 'actions']);

  const refs = {
    animatedViewRef: useRef<View>(null),
  };

  const handleSetNotFoundProduct = useCallback(() => {
    actions.CLEAR_PRODUCT_NOT_FOUND();
  }, [actions]);

  useEffect(() => {
    if (refs.animatedViewRef.current) {
      if (refs.animatedViewRef.current.slideInDown) {
        refs.animatedViewRef.current.slideInDown();
      }
    }
  }, [refs.animatedViewRef]);

  return (
    <View
      ref={refs.animatedViewRef}
      testID="com.usereserva:id/NotFoundProduct_container"
      animation="slideInDown"
      useNativeDriver
      style={{
        elevation: 10,
        position: 'absolute',
        right: 0,
        left: 0,
        zIndex: 2,
      }}
    >
      <Box
        minHeight={60}
        bg="white"
        paddingLeft="xxxs"
        py="micro"
        flexDirection="row"
        alignItems="center"
        paddingRight="xxxs"
        boxShadow={Platform.OS === platformType.IOS ? 'topBarShadow' : null}
        style={{ elevation: 10 }}
      >
        <Box flex={1}>
          <Typography
            fontFamily="nunitoRegular"
            fontSize={13}
            color="preto"
          >
            {`${productNotFound}`}
          </Typography>
        </Box>
        <Button testID="com.usereserva:id/NotFoundProduct_setProduct" flex={1} onPress={handleSetNotFoundProduct}>
          <IconLegacy name="Close" size={15} color="preto" ml="xxxs" />
        </Button>
      </Box>
    </View>
  );
}
