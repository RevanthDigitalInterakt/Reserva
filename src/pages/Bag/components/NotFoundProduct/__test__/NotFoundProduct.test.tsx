import React from 'react';
import { theme } from '@usereservaapp/reserva-ui';
import { ThemeProvider } from 'styled-components/native';
import { fireEvent, render, screen } from '@testing-library/react-native';
import NotFoundProduct from '..';
import * as useBagStore from '../../../../../zustand/useBagStore/useBagStore';

const mockClearProductNotFound = jest.fn();

jest.spyOn(useBagStore, 'useBagStore').mockReturnValue({
  actions: {
    CLEAR_PRODUCT_NOT_FOUND: mockClearProductNotFound,
  },
  productNotFound: 'Product not found',
} as any);

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <NotFoundProduct />
  </ThemeProvider>
);

describe('NotFoundProduct component', () => {
  beforeEach(() => {
    render(TestingComponent);
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match with the snapshot', () => {
    const notFoundProductComponent = screen.toJSON();

    expect(notFoundProductComponent).toMatchSnapshot();
  });

  it('should render with calls handleSetNotFoundProduct when close button is pressed', async () => {
    const closeButton = screen.getByTestId('com.usereserva:id/NotFoundProduct_setProduct');

    fireEvent.press(closeButton);

    expect(mockClearProductNotFound).toHaveBeenCalled();
  });
});
