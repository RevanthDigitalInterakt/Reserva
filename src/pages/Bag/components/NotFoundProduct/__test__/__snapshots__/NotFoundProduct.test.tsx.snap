// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundProduct component should match with the snapshot 1`] = `
<View
  animation="slideInDown"
  style={
    {
      "elevation": 10,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "zIndex": 2,
    }
  }
  testID="com.usereserva:id/NotFoundProduct_container"
  useNativeDriver={true}
>
  <View
    alignItems="center"
    bg="white"
    boxShadow="topBarShadow"
    flexDirection="row"
    minHeight={60}
    paddingLeft="xxxs"
    paddingRight="xxxs"
    py="micro"
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "flexDirection": "row",
          "minHeight": 60,
          "paddingBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12,
          "shadowColor": "rgba(0,0,0,0.1)",
          "shadowOffset": {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
        },
        {
          "elevation": 10,
        },
      ]
    }
  >
    <View
      flex={1}
      style={
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <Text
        color="preto"
        fontFamily="nunitoRegular"
        fontSize={13}
        style={
          {
            "color": "#333333",
            "fontFamily": "Nunito-Regular",
            "fontSize": 13,
          }
        }
      >
        Product not found
      </Text>
    </View>
    <View
      alignItems="center"
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.usereserva:id/NotFoundProduct_setProduct"
      >
        <RNSVGSvgView
          bbHeight="15"
          bbWidth="15"
          color="preto"
          focusable={false}
          ml="xxxs"
          size={15}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#333333",
                "height": 15,
                "marginLeft": 16,
                "width": 15,
              },
              {
                "flex": 0,
                "height": 15,
                "width": 15,
              },
            ]
          }
          tintColor="preto"
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="100%"
              bbWidth="100%"
              focusable={false}
              height="100%"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "height": "100%",
                    "width": "100%",
                  },
                  {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={8.24}
              vbWidth={8.242}
              width="100%"
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGPath
                  d="M5.098 4.12l2.944-2.944A.69.69 0 007.066.2L4.122 3.144 1.178.2a.69.69 0 10-.976.976L3.146 4.12.202 7.064a.69.69 0 00.976.976l2.944-2.944L7.066 8.04a.69.69 0 00.976-.976z"
                  fill={
                    {
                      "payload": 4281545523,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
